src/state/nftMarket/reducer.ts

```diff
@@ -50,6 +66,34 @@ export const fetchNftsFromCollections = createAsyncThunk<NFT[], string>(
   },
 )
 
+export const fetchUserNfts = createAsyncThunk<
+  NftSubgraphEntity[],
+  { account: string; profileNftWithCollectionAddress: TokenIdWithCollectionAddress; collections: ApiCollections }
+>('nft/fetchUserNfts', async ({ account, profileNftWithCollectionAddress, collections }) => {
+  const nftsInWallet = await fetchWalletTokenIdsForCollections(account, collections)
+
+  if (profileNftWithCollectionAddress?.tokenId) {
+    nftsInWallet.push(profileNftWithCollectionAddress)
+  }
+
+  const nftIdsInWallet = nftsInWallet.map((nft) => nft.tokenId)
+  const marketDataForNftsInWallet = await getNftsMarketData({ tokenId_in: nftIdsInWallet })
+  const walletNftsWithMarketData = nftsInWallet.map((walletNft) => {
+    const marketData = marketDataForNftsInWallet.find((marketNft) => marketNft.tokenId === walletNft.tokenId)
+    return (
+      marketData || {
+        tokenId: walletNft.tokenId,
+        collectionAddress: walletNft.collectionAddress,
+        metadataUrl: null,
+        transactionHistory: null,
+      }
+    )
+  })
+
+  const marketDataForNftsOnSale = await getNftsMarketData({ currentSeller: account.toLowerCase() })
+  return [...walletNftsWithMarketData, ...marketDataForNftsOnSale]
```

Comment:
This returns very simple data for now until we have the API and know more about how collection metadata is retrieved.

It's just an array of objects with a `tokenId`, `collectionAddress`, a `metaDataUrl` and a `transactionHistory` object.